src/IntervalTrees.jl

```diff
@@ -641,11 +637,11 @@ function _push!(t::IntervalBTree{K, V, B},
         child = leftleaf
         rightnode = rightleaf
         hassplit = true
-        while !isnull(parent)
-            p = get(parent)
+        while parent !== nothing
+            p = parent
             if hassplit
                 i = findidx(p, entry) # key index
-                j = i <= length(p) - 1 && entry >= p.keys[i] ? i + 1 : i # child index
+                j = i <= length(p) - 1 && entry < p.keys[i] ? i : i + 1 # child index
```

Comment:
Please review this code in light of #36. While this is just a workaround a complete elaborate fix will needed as described in #36.